control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go

```diff
@@ -1,209 +1,316 @@
-				},
-			},
+		{
+			Name:       "etcd-client",
+			Protocol:   corev1.ProtocolTCP,
+			Port:       2379,
+			TargetPort: intstr.Parse("client"),
 		},
 	}
-	deploymentConfig.ApplyTo(deployment)
 	return nil
 }
 
-func ReconcileCluster(cluster *etcdv1.EtcdCluster, ownerRef config.OwnerRef, etcdDeploymentConfig config.DeploymentConfig, etcdVersion string, pvcClaim *corev1.PersistentVolumeClaimSpec) error {
-	ownerRef.ApplyTo(cluster)
-	peerSecret := manifests.EtcdPeerSecret(cluster.Namespace)
-	serverSecret := manifests.EtcdServerSecret(cluster.Namespace)
-	clientSecret := manifests.EtcdClientSecret(cluster.Namespace)
+func ReconcileClientService(service *corev1.Service, ownerRef config.OwnerRef) error {
+	if service.CreationTimestamp.IsZero() {
+		service.Spec.Type = corev1.ServiceTypeClusterIP
+		service.Spec.ClusterIP = corev1.ClusterIPNone
+	}
+	ownerRef.ApplyTo(service)
+	service.Labels = etcdPodSelector
+	service.Spec.Selector = etcdPodSelector
+	service.Spec.Ports = []corev1.ServicePort{
+		{
+			Name:       "etcd-client",
+			Protocol:   corev1.ProtocolTCP,
+			Port:       2379,
+			TargetPort: intstr.Parse("client"),
+		},
+		{
+			Name:       "metrics",
+			Protocol:   corev1.ProtocolTCP,
+			Port:       2381,
+			TargetPort: intstr.Parse("metrics"),
+		},
+	}
+	return nil
+}
 
-	podPolicy := &etcdv1.PodPolicy{}
+func ReconcileServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(sm)
 
-	if resources, ok := etcdDeploymentConfig.Resources[etcdContainer().Name]; ok {
-		podPolicy.Resources = resources
+	sm.Spec.Selector.MatchLabels = etcdPodSelector
+	sm.Spec.NamespaceSelector = prometheusoperatorv1.NamespaceSelector{
+		MatchNames: []string{sm.Namespace},
 	}
-	podPolicy.Affinity = etcdDeploymentConfig.Scheduling.Affinity
-	podPolicy.Tolerations = etcdDeploymentConfig.Scheduling.Tolerations
-	podPolicy.PriorityClassName = etcdDeploymentConfig.Scheduling.PriorityClass
-	if sc, ok := etcdDeploymentConfig.SecurityContexts[etcdContainer().Name]; ok {
-		podPolicy.SecurityContext = &corev1.PodSecurityContext{
-			SELinuxOptions: sc.SELinuxOptions,
-			WindowsOptions: sc.WindowsOptions,
-			RunAsUser:      sc.RunAsUser,
-			RunAsGroup:     sc.RunAsGroup,
-			RunAsNonRoot:   sc.RunAsNonRoot,
-			SeccompProfile: sc.SeccompProfile,
-		}
-	}
-	podPolicy.PersistentVolumeClaimSpec = pvcClaim
-	podPolicy.Labels = etcdDeploymentConfig.AdditionalLabels
-
-	cluster.Spec = etcdv1.ClusterSpec{
-		Size:    etcdDeploymentConfig.Replicas,
-		Version: etcdVersion,
-		TLS: &etcdv1.TLSPolicy{
-			Static: &etcdv1.StaticTLS{
-				Member: &etcdv1.MemberSecret{
-					PeerSecret:   peerSecret.Name,
-					ServerSecret: serverSecret.Name,
+	sm.Spec.Endpoints = []prometheusoperatorv1.Endpoint{
+		{
+			Interval: "15s",
+			Port:     "metrics",
+			Scheme:   "https",
+			TLSConfig: &prometheusoperatorv1.TLSConfig{
+				SafeTLSConfig: prometheusoperatorv1.SafeTLSConfig{
+					InsecureSkipVerify: true,
```

Comment:
Good catch, might be a leftover from prototyping. I'll investigate.

Reply:
Now I remember:
```
Get "https://10.131.0.115:2381/metrics": x509: cannot validate certificate for 10.131.0.115 because it doesn't contain any IP SANs
```
All of the certs are signed for etcd member DNS names, need to figure out how to account for it